### ISSUE TITLE
Remove external-attacher from the repo.

### ISSUE THREAD

**Issue by humblec** [ceph/ceph-csi#1106]
# Describe the bug #

The presence of external-attacher adds on overhead and issues wrt attachment in various scenarios. One of them would be the lack of performance on syncing volumeattachment from api server..etc. More or less we don't have controller publish and unpublish capabilities , so we should not make use of this sidecar and cause unnecessary addon here thus other issues. 

# Environment details #

- Image/version of Ceph CSI driver :
- Helm chart version :
- Kernel version :
- Mounter used for mounting PVC (for cephfs its `fuse` or `kernel`. for rbd its
  `krbd` or `rbd-nbd`) :
- Kubernetes cluster version :
- Ceph cluster version :

# Steps to reproduce #

1. Setup details: '...'
1. Deployment to trigger the issue '....'
1. See error

# Actual results #

Describe what happened

# Expected behavior #

# Logs #

If the issue is in PVC creation, deletion, cloning please attach complete logs
of below containers.

- csi-provisioner and csi-rbdplugin/csi-cephfsplugin container logs from the
  provisioner pod.

If the issue is in PVC resize please attach complete logs of below containers.

- csi-resizer and csi-rbdplugin/csi-cephfsplugin container logs from the
  provisioner pod.

If the issue is in snapshot creation and deletion please attach complete logs
of below containers.

- csi-snapshotter and csi-rbdplugin/csi-cephfsplugin container logs from the
  provisioner pod.

If the issue is in PVC mounting please attach complete logs of below containers.

- csi-rbdplugin/csi-cephfsplugin and driver-registrar container logs from
  plugin pod from the node where the mount is failing.

- if required attach dmesg logs.

**Note:-** If its a rbd issue please provide only rbd related logs, if its a
cephfs issue please provide cephfs logs.

# Additional context #

For example:

Any existing bug report which describe about the similar issue/behavior

**Comment by Madhu-1** [comment]
This needs extensive testing to make sure there won't be an issue for RWO PVC

**Comment by Madhu-1** [comment]
this is already tried in https://github.com/rook/rook/pull/4172 and reverted back https://github.com/rook/rook/pull/4332 due to the issue we have seen in RWO PVC

**Comment by humblec** [comment]
> This needs extensive testing to make sure there won't be an issue for RWO PVC

Yeah, rather this need extensive testing for all types of PVCs :).

**Comment by humblec** [comment]
> this is already tried in [rook/rook#4172](https://github.com/rook/rook/pull/4172) and reverted back [rook/rook#4332](https://github.com/rook/rook/pull/4332) due to the issue we have seen in RWO PVC

The difference here is, this is GA in 1.18 compared to previous versions. So ideally it should work without issues. Lets try out with 1.18 version and see.

**Comment by ShyamsundarR** [comment]
Tagging @gnufied 

On a prior discussion regarding removal of the attacher, as it improved performance of overall pod startup times, it was pointed out by @gnufied that this essentially means double attach is detected and prevented by the storage plugin, and the basic AD controller will not ensure that the older node running the pod has actually detached etc. (this is even when the csi plugin ControllerPublish/Unpublish are NOPs) (unable to find a link to the discussion as yet)

Also, as @Madhu-1 points out, the prior attempt led to the related problems.

A subsequent speed up of the attach-detach operation was attempted [here](https://bugzilla.redhat.com/show_bug.cgi?id=1760940) and was not fully successful.

HTH, in taking this forward as appropriate.

**Comment by Madhu-1** [comment]
@humblec are you working on the fix? would you like to track this one for v3.0.0 if not please move this out of v3.0.0 release milestone

**Comment by Madhu-1** [comment]
Moving it to release-v3.1.0

**Comment by Madhu-1** [comment]
@humblec can this be done in 3.1.0? or do we need to move it out to the next release?

**Comment by humblec** [comment]
Yeah, we have to move this from 3.1.0. Done.. thanks :+1:

**Comment by humblec** [comment]
Considering `skipAttach` has been GAd with `1.18` and take care of the burden on unncessary operation (round-trip) in the preparation of a volume for a container, and requires CSI Drivers to deploy an unnecessary sidecar container (external-attacher), this is a good time to take it out.
